thirdeye/thirdeye-frontend/app/pods/sandbox/controller.js

```diff
@@ -0,0 +1,136 @@
+import Ember from 'ember';
+import moment from 'moment';
+import fetch from 'fetch';
+
+const generateRandomData = (num, min=0, max=100) => {
+  const diff = max - min;
+  const dataPoints = [];
+  for (let i = 0; i < num; i++) {
+    dataPoints.push(Math.random() * diff + min);
+  }
+  return dataPoints;
+};
+
+const generateSequentialData = (num, min=0, max=100) => {
+  const diff = max - min;
+  const dataPoints = [];
+  for (let i = 0; i < num; i++) {
+    dataPoints.push(min + diff * (i / num));
+  }
+  return dataPoints;
+};
+
+export default Ember.Controller.extend({
+
+  types: null,
+
+  colors: null,
+
+  axes: null,
+
+  series: Ember.computed(
+    'model.data',
+    'types',
+    'colors',
+    'axes',
+    function() {
+      console.log('series()');
+
+      const types = this.get('types') || {};
+      const colors = this.get('colors') || {};
+      const axes = this.get('axes') || {};
+      const data = this.get('model.data') || {};
+      console.log('data', data);
+
+      const series = {};
+      Object.keys(data).forEach(range =>
+        Object.keys(data[range]).filter(s => s != 'timestamp').forEach(mid => {
+          const sid = range + '-' + mid;
+          series[sid] = {
+            timestamps: data[range]['timestamp'],
+            values: data[range][mid],
+          };
+
+          if (sid in types) {
+            series[sid].type = types[sid];
+          }
+
+          if (sid in colors) {
+            series[sid].color = colors[sid];
+          }
+
+          if (sid in axes) {
+            series[sid].axis = axes[sid];
+          }
+      }));
+
+      console.log('series', series);
+      return series;
+    }
+  ),
+
+  actions: {
+    addSeries(sid) {
+      console.log('addSeries()');
+
+      const data = this.get('model.data');
+
+      data['myrange'] = {
+        [sid]: generateRandomData(100, 0, 20000),
+        "timestamp": generateSequentialData(100, 1508454000000, 1508543940000)
+      };
+
+      console.log('data', data);
+      this.set('model.data', data);
+      this.notifyPropertyChange('model');
```

Comment:
do we need to use ` this.notifyPropertyChange('model');`, it's hard to determine what this will trigger. I'd rather have a computed property on `'model', 'model.data'`

Reply:
Applies to other actions

Reply:
deleted